packages/extras/src/lib/hooks/useFBO.ts

```diff
@@ -0,0 +1,67 @@
+/* Based on https://github.com/pmndrs/drei/blob/master/src/core/useFBO.tsx under the MIT License */
+import { useThrelte } from '@threlte/core'
+import { onDestroy, onMount } from 'svelte'
+import * as THREE from 'three'
+
+type UseFBOOptions = {
+  /** Defines the count of MSAA samples. Can only be used with WebGL 2. Default: 0 */
+  samples?: number
+  /** If set, the scene depth will be rendered into buffer.depthTexture. Default: false */
+  depth?: boolean
+} & THREE.WebGLRenderTargetOptions
+
+// 👇 uncomment when TS version supports function overloads
+// export function useFBO(options?: UseFBOOptions)
+export function useFBO(
+  /** Width in pixels, or options (will render fullscreen by default) */
+  width?: number | UseFBOOptions,
+  /** Height in pixels */
+  height?: number,
+  /** Options */
+  options?: UseFBOOptions
+): THREE.WebGLRenderTarget {
+  const { dpr, size } = useThrelte()
+
+  const _width = typeof width === 'number' ? width : 0
+  const _height = typeof height === 'number' ? height : 0
+  const _options = (typeof width === 'number' ? options : (width as UseFBOOptions)) || {}
+  const { samples = 0, depth, ...targetOptions } = _options
+
+  const target = new THREE.WebGLRenderTarget(_width, _height, {
+    minFilter: THREE.LinearFilter,
+    magFilter: THREE.LinearFilter,
+    type: THREE.HalfFloatType,
+    ...targetOptions
+  })
+
+  if (depth) {
+    target.depthTexture = new THREE.DepthTexture(_width, _height, THREE.FloatType)
+  }
+
+  target.samples = samples
+
+  onMount(() => {
+    if (samples) target.samples = samples
+  })
+
+  onDestroy(() => {
+    target.dispose()
+  })
+
+  const unsubscribeSize = size.subscribe((val) => {
+    // Update the width and height on size change
+    const _width = typeof width === 'number' ? width : val.width * dpr.current
+    const _height = typeof height === 'number' ? height : val.height * dpr.current
+    if (target.width !== _width && target.height !== _height) {
+      target.setSize(_width, _height)
+      if (depth) {
+        target.depthTexture?.dispose()
```

Comment:
I'm not sure if this section with recreating the `DepthTexture` is strictly necessary. It would be great to get a second pair of eyes. @grischaerbe

Reply:
Here from review ping but I'd prefer to not make a call on this. Happy with the direction we've talked about for the docs part tho :+1:

Reply:
It seems the drei implementation does leave that out, so I would follow their lead on this 👍

Reply:
Something that's been on my mind for this is the difference between drei's
```jsx
const target = React.useMemo(() => {
    const target = new THREE.WebGLRenderTarget(_width, _height, {
      minFilter: THREE.LinearFilter,
      magFilter: THREE.LinearFilter,
      type: THREE.HalfFloatType,
      ...targetSettings,
    })
    // [...]
    return target
  }, [])
  ```
  
  and here with
  
  ```js
  const target = new THREE.WebGLRenderTarget(_width, _height, {
    minFilter: THREE.LinearFilter,
    magFilter: THREE.LinearFilter,
    type: THREE.HalfFloatType,
    ...targetOptions
  })
  ```
  
Having it reactive would be more aligned with drei but I don't know whether or not the reactivity is something to worry about. @ivoilic is this something you come shed some light on? I could just be being dumb here.

Reply:
@DefinitelyMaybe If I'm not mistaken, this is not reactive. The dependency array is empty, meaning that by react laws it will only be created once on component initialization.

Reply:
Ah! Good 👍 no worries there then.

Also kinda strange of drei to put that in then... Tho again that might just be my lack of knowledge about react.

Reply:
@grischaerbe Per your recommendation the depth texture recreation is now removed in my most recent commit [128c1df](https://github.com/threlte/threlte/pull/815/commits/128c1df443ae255820993f19a04b4ef94b21ab07)